numpy/core/src/umath/loops.c.src

```diff
@@ -540,6 +540,24 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ void
         BINARY_LOOP_FAST(@type@, @type@, *out = in1 @OP@ in2);
     }
 }
+
+NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ int
+@TYPE@_@kind@@isa@_indexed(PyArrayMethod_Context *NPY_UNUSED(context), char * const*args, npy_intp const *dimensions, npy_intp const *steps, NpyAuxData *NPY_UNUSED(func))
+{
+    char *ip1 = args[0];
+    npy_intp *indx = (npy_intp *)args[1];
+    char *value = args[2];
+    npy_intp is1 = steps[0], is2 = steps[1] / sizeof(npy_intp), os1 = steps[2];
```

Comment:
Total nitpick, but I'd call the step sizes something a bit closer to the description for `np.ufunc.at`, e.g., `isa`, `isindex`, and `isb` -- the use of "os1" for `value` suggests an output.

I'm not quite sure about the cast here for `is2`, since the compiler will just implicitly change it by the same factor in the loop anyway.

Reply:
I will change the naming in all the loops. 
>I'm not quite sure about the cast here for is2

Which cast?  `steps[1]` is in bytes, but since `indx` is already cast to `npy_intp`, `steps[1]` must be ajusted to `sizeof(npy_intp)`.

Reply:
fixing to have `char * indx = args[1]`